packages/openactive-integration-tests/test/features/core/amending-order-quote/implemented/amend-c1-test.js

```diff
@@ -0,0 +1,106 @@
+  // The secondary opportunity criteria to use for multiple OrderItem tests
+  controlOpportunityCriteria: 'TestOpportunityBookable',
+  // This test uses 2 opportunities, A & B
+  numOpportunitiesUsedPerCriteria: 2,
+},
+(configuration, orderItemCriteria, featureIsImplemented, logger) => {
+  // Both runs share the same UUID, so that the 2nd run is an amendment to the same Order
+  const uuid = generateUuid();
+  /**
+   * Note: This generates an it() block. Therefore, this must be run within a describe() block.
+   *
+   * @param {RequestState} state
+   * @param {C1 | B} stage
+   * @param {() => any} responseAccessorFn function that gets the stage's response (e.g. `() => state.c1Response`)
+   */
+  const itFeedItemAndResponseItemShouldMatchIds = (state, stage, responseAccessorFn) => {
+    Common.itForOrderItem(orderItemCriteria, state, stage, () => responseAccessorFn().body,
+      'ID should match the one specified in the open data feed',
+      (feedOrderItem, responseOrderItem) => {
+        chai.expect(responseOrderItem).to.nested.include({
+          'orderedItem.@id': feedOrderItem.orderedItem['@id'],
+        });
+      });
+  };
+  // N.B.: The following two tests must be performed sequentially - with
+  // Second Attempt occurring after First Attempt.
+  describe('First Attempt - C1', () => {
+    // Each of these scenarios uses a separate state and flowHelper because they fetch separate opportunities
+    const state = new RequestState(logger, uuid);
+    const flow = new FlowHelper(state);
+
+    beforeAll(async () => {
+      await state.fetchOpportunities(orderItemCriteria);
+      await chakram.wait();
+    });
+
+    (new GetMatch({
+      state, flow, logger, orderItemCriteria,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+    const c1 = (new C1({
+      state, flow, logger,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+
+    itFeedItemAndResponseItemShouldMatchIds(state, c1, () => state.c1Response);
+  });
+  /** Fetch some new opportunities and amend the existing order at C1, and then complete it */
+  describe('Second Attempt - C1 -> B', () => {
+    // Each of these scenarios uses a separate state and flowHelper because they fetch separate opportunities
+    const state = new RequestState(logger, uuid);
+    const flow = new FlowHelper(state);
+
+    beforeAll(async () => {
+      await state.fetchOpportunities(orderItemCriteria);
+      await chakram.wait();
+    });
+
+    (new GetMatch({
+      state, flow, logger, orderItemCriteria,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+    const c1 = (new C1({
+      state, flow, logger,
+    }))
+      .beforeSetup()
+      .successChecks()
+      .validationTests();
+
+    // This should verify that the order has now been amended
+    itFeedItemAndResponseItemShouldMatchIds(state, c1, () => state.c1Response);
+
+    const b = (new B({
```

Comment:
C2 is required before calling B...

Perhaps this test should actually go C1 -> C2 -> C1 -> C2 -> B?

Reply:
> C2 is required before calling B...

Yes. The flow helper calls C2 if it hasn't already been called though. Are you saying that it's better to be explicit and call it in the test?

> Perhaps this test should actually go C1 -> C2 -> C1 -> C2 -> B?

Are you saying that the 1st attempt should call C2 after C1? In the spreadsheet, that's a separate test

Reply:
> flow helper calls C2 if it hasn't already been called
I wasn't aware of this, in which case no worries!

And sorry I should have checked the sheet, yes a separate test would indeed be better, and is needed if it's already in the sheet.

Reply:
I've also made the C2 call explicit. I agree that we should move towards the tests being understandable even if you don't understand the framework. This also means that changes to the framework have less destabilising effects on the tests